code/modules/food_and_drinks/drinks/drinks/modglass.dm

```diff
@@ -0,0 +1,202 @@
+	name = "malleable glass"
+	desc = "Not your standard drinking glass!"
+	icon = 'icons/obj/food/modglass.dmi'
+	icon_state = "mglass-1-"
+	fill_icon = 'icons/obj/food/modglass_fillings.dmi'
+	fill_icon_thresholds = list(25,50)
+	amount_per_transfer_from_this = 10
+	volume = 50
+	custom_materials = list(/datum/material/glass=500, /datum/material/silver=100)
+	max_integrity = 30
+	spillable = TRUE
+	resistance_flags = ACID_PROOF
+	obj_flags = UNIQUE_RENAME
+	drop_sound = 'sound/items/handling/drinkglass_drop.ogg'
+	pickup_sound =  'sound/items/handling/drinkglass_pickup.ogg'
+	custom_price = 25
+	//rim defines the size of rim the glass has, used to decide which skins are available, and which garnish sprites to use
+	var/rim = RIM_MEDIUM
+	//stores the number of variations this glass sprite has to select from
+	var/variants = MEDIUM_VARIANTS
+	//a list to be filled with the associative list containing possible skins
+	var/list/glass_skins = list()
+	//a list to be filled with the current garnishes placed on the glass
+	var/list/garnishes = list()
+
+/obj/item/reagent_containers/food/drinks/modglass/small
+	name = "small malleable glass"
+	icon_state = "sglass-1-"
+	custom_materials = list(/datum/material/glass=100, /datum/material/silver=100)
+	volume = 25
+	rim = RIM_SMALL
+	variants = SMALL_VARIANTS
+
+/obj/item/reagent_containers/food/drinks/modglass/large
+	name = "large malleable glass"
+	icon_state = "lglass-1-"
+	rim = RIM_LARGE
+	variants = LARGE_VARIANTS
+
+/obj/item/reagent_containers/food/drinks/modglass/Initialize()
+	. = ..()
+	if(variants)
+		glass_skins = glass_variants_list()
+
+//steals code from tile_reskinning.dm to cache an associative list containing possible reskins of the glass
+/obj/item/reagent_containers/food/drinks/modglass/proc/glass_variants_list()
+	. = GLOB.glass_variants[rim]
+	if(.)
+		return
+	for(var/variant in 1 to variants)
+		var/name_string = "[rim]glass-[variant]-"
+		glass_skins[name_string] = icon('icons/obj/food/modglass.dmi', "[name_string]")
+	return GLOB.glass_variants[rim] = glass_skins
+
+//if this glass can be reskinned, open a radial menu containing the skins, and change the icon_state to whatever is chosen
+/obj/item/reagent_containers/food/drinks/modglass/AltClick(mob/user)
+	if(!glass_skins)
+		return
+	var/choice = show_radial_menu(user, src, glass_skins, radius = 48, require_near = TRUE)
+	if(!choice || choice == icon_state)
+		return
+	icon_state = choice
+	update_icon()
+
+//if the object is a garnish, with a valid garnish_state, and there isnt already a garnish of the same type, add it to the list at the index of its layer
+/obj/item/reagent_containers/food/drinks/modglass/attackby(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/garnish))
+		return ..()
+	var/obj/item/garnish/item = I
+	if(!item.garnish_state)
+		return ..()
+	if(garnishes["[item.garnish_layer]"])
+		to_chat(user, "<span class='notice'>You can't fit [item] on the glass!</span>")
```

Comment:
something like "theres already an item on this layer" but better sounding would tell more to players